blocks/form/submit.js

```diff
@@ -59,42 +60,39 @@ function getFieldValue(fe, payload) {
 }
 
 function constructPayload(form) {
-  const payload = { __id__: generateUnique() };
+  const formData = new FormData();
   [...form.elements].forEach((fe) => {
     if (fe.name && !fe.matches('button') && !fe.disabled && fe.tagName !== 'FIELDSET') {
-      const value = getFieldValue(fe, payload);
+      const value = getFieldValue(fe, formData);
       if (fe.closest('.repeat-wrapper')) {
-        payload[fe.name] = payload[fe.name] ? `${payload[fe.name]},${fe.value}` : value;
+        formData.append(fe.name, value);
       } else {
-        payload[fe.name] = value;
+        formData.set(fe.name, value);
       }
     }
   });
-  return { payload };
+  return formData;
 }
 
 async function prepareRequest(form) {
-  const { payload } = constructPayload(form);
-  const headers = {
-    'Content-Type': 'application/json',
-  };
-  const body = { data: payload };
-  const url = form.dataset.submit || form.dataset.action;
-  return { headers, body, url };
+  const formData = constructPayload(form);
+  const url = `${getSubmitBaseUrl()}${formData.get('xfpath')}.form`;
+  return { formData, url };
 }
 
 export async function submitForm(form, captcha) {
   try {
-    const { headers, body, url } = await prepareRequest(form, captcha);
+    // eslint-disable-next-line prefer-const
+    let { formData, url } = await prepareRequest(form, captcha);
     let token = null;
     if (captcha) {
       token = await captcha.getToken();
-      body.data['g-recaptcha-response'] = token;
+      formData.append('g-recaptcha-response', token);
+      url += `?g-recaptcha-response=${encodeURIComponent(token)}`;
```

Comment:
We are already including it as part of form data do we still need it as part of query parameter?

Reply:
Not required so removed it!